ext/Logger.php

```diff
@@ -63,26 +65,52 @@ public function _initialize()
         if (!class_exists('\Monolog\Logger')) {
             throw new ConfigurationException("Logger extension requires Monolog library to be installed");
         }
-        $this->path = $this->getLogDir();
+    }
+
+    public static function getLogger()
+    {
+        if (self::$logger) {
+            $logger = self::$logger;
+        } else {
+            $logger = self::getDefaultLogger();
```

Comment:
Also I think logger should not be created for every purpose. It's ok if it would be created one time at extension start.
So `getLogger` should only return already created logger and not try to recreate it.

Here is the point. The logger should be accessible only if extension is loaded and initialize. If it's not enabled in config and `getLogger` still works that can be confusing.